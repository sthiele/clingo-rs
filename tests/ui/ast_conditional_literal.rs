use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let mut lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    let term2 = ast::Term::from(sym);
    let lit2 = ast::Literal::from_term(ast::Sign::NoSign, &term2);
    let args = vec![lit2];
    let cond = ast::ConditionalLiteral::new(&lit, &args);
    lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
    drop(args);
    let _end = (lit, cond);
}