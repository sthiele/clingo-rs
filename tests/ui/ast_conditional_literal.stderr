error[E0506]: cannot assign to `lit` because it is borrowed
  --> $DIR/ast_conditional_literal.rs:11:5
   |
10 |     let cond = ast::ConditionalLiteral::new(&lit, &args);
   |                                             ---- borrow of `lit` occurs here
11 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `lit` occurs here
12 |     drop(args);
13 |     let _end = (lit, cond);
   |                      ---- borrow later used here

error[E0505]: cannot move out of `args` because it is borrowed
  --> $DIR/ast_conditional_literal.rs:12:10
   |
10 |     let cond = ast::ConditionalLiteral::new(&lit, &args);
   |                                                   ----- borrow of `args` occurs here
11 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term1);
12 |     drop(args);
   |          ^^^^ move out of `args` occurs here
13 |     let _end = (lit, cond);
   |                      ---- borrow later used here

Some errors have detailed explanations: E0505, E0506.
For more information about an error, try `rustc --explain E0505`.
